### PR TITLE
Add a detailed overview of transports to initial readme

### PR DESCRIPTION
**Pull request by jsumners**
See issue #263 .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,6 +140,45 @@ For a fair comparison, [LogLevel](http://npm.im/loglevel) was extended
...
+
+A transport in most logging libraries is something that runs in-process to
+perform some operation with the finalized log line. For example, a transport
+might send the log line to a standard syslog server after processing the log
+line and reformatting it. For details on implementing, and some already written,
+transports, see our [Transports⇗](docs/transports.md) document.
+
+> **Pino *does not* natively support in-process transports.**
+
+Pino does not support in-process transports because Node processes are
```
Comment (davidmarkclements) [inline README.md]: this isn't true - we do now (multi stream)

File: README.md [hunk 2]
```diff
@@ -140,6 +140,45 @@ For a fair comparison, [LogLevel](http://npm.im/loglevel) was extended
...
+single threaded processes (ignoring some technical details). In order for Pino
+to achieve the speed it achieves, Pino purposefully offloads the handling of
```
Comment (davidmarkclements) [inline README.md]: This  indicates that's how Pino achieves high speed - it isn't. 

It's rather that we recommend out-of-process as the optimum

File: README.md [hunk 3]
```diff
@@ -140,6 +140,45 @@ For a fair comparison, [LogLevel](http://npm.im/loglevel) was extended
...
+logs, and their ultimate destination, to external processes so that the
+threading capabilities of the OS can be used (or other CPUs).
+
+One consequence of this methodology is that "error" logs do not get written to
```
Comment (davidmarkclements) [inline README.md]: this is normative for log4j, we're a log4j logger, so while we should mention this I don't think it should be skewed by other peoples misunderstandings of log4j - can we elaborate that it's because of this *and* log4j that all logs go to the same destination stream (as set by the user, defaulting to stdout)

File: README.md [hunk 4]
```diff
@@ -140,6 +140,45 @@ For a fair comparison, [LogLevel](http://npm.im/loglevel) was extended
...
+`stderr`, as might be exepected. However, since Pino logs are in a parseable
```
Comment (davidmarkclements) [inline README.md]: typo: expected

Comment (davidmarkclements) [inline README.md]: probably would avoid saying it's "as might be expected" 
 
whilst it might be, it's not an expectation based on reasonable assumption, rather its a by product of familiarity bias

File: README.md [hunk 5]
```diff
@@ -140,6 +140,45 @@ For a fair comparison, [LogLevel](http://npm.im/loglevel) was extended
...
+```
+$ node an-app.js | jq 'select(.level == 50)'
+```
+
+In short, the way Pino generates logs:
+
+1. Reduces the impact of logging on your application to an extremely minimal amount.
+2. Gives greater flexibility in how logs are processed and stored.
+
+Given all of the above, Pino clearly promotes out-of-process log processing.
```
Comment (davidmarkclements) [inline README.md]: should we create two headings - one for transports and the other for log processing?

File: README.md (hunk 1 above)
Comment (jsumners) [inline README.md]: That requires a separate package. Out of the box, `npm install pino`, we don't deal with in-process transports.

File: README.md (hunk 2 above)
Comment (jsumners) [inline README.md]: Hmm, I meant to imply that it is one of the factors in Pino's speed. How about "...details). Given this restriction, one of the methods Pino employs to achieve its speed is to purposfully offload..."?

File: README.md (hunk 3 above)
Comment (jsumners) [inline README.md]: I specifically left log4j out of this because what you are saying about log4j isn't actually true. We only follow the API it defines (`.error`, `.debug`, etc). But that's only half(?) of log4j. The other part of log4j is that it handles log management. Look at the log4j configuration for Elastic Search -- https://github.com/elastic/elasticsearch/blob/847378a43b066b33f731e4948589f6040f69d2f6/distribution/src/main/resources/config/log4j2.properties . All of that mess is configuring the log4j library to write the combined stream out to various destinations. So by saying we conform to the log4j standard, we'd be suggesting that we should support writing to multiple streams by default.

File: README.md (hunk 4 above)
Comment (jsumners) [inline README.md]: I'll strike the clause.

File: README.md (hunk 5 above)
Comment (jsumners) [inline README.md]: I'm trying to directly respond to questions around "transports" that have been raised multiple times. So I'm trying to give strong reasons for the way things are. Does that make sense?

File: README.md (hunk 1 above)
Comment (mcollina) [inline README.md]: @jsumners is correct. But we should mention multi-stream.

File: README.md (hunk 2 above)
Comment (mcollina) [inline README.md]: This is the optimimum for a lot of reasons, including performance. Logs are better sent somewhere else for handling, maybe even docker, or somewhere else. Definitely transports do not fit the single-process module that Node.js has.

File: README.md (hunk 3 above)
Comment (mcollina) [inline README.md]: I'm very -1 on supporting multiple streams by default, even though that is possible.

I'm -1 on mentioning log4j, however we should include a little compatibility table somewhere, discussing how far we are from the other loggers API.

File: README.md (hunk 5 above)
Comment (mcollina) [inline README.md]: I'm ok on the way @jsumners wrote it.

File: README.md (hunk 3 above)
Comment (jsumners) [inline README.md]: @mcollina I think such a table should be a separate PR.

Comment (mcollina) [inline README.md]: 👍

File: README.md (hunk 5 above)
Comment (davidmarkclements) [inline README.md]: ok 👍

File: README.md (hunk 2 above)
Comment (jsumners) [inline README.md]: So is my proposed "fix" acceptable?

Comment (davidmarkclements) [inline README.md]: :+1:

**Comment by jsumners** [comment]
@davidmarkclements I believe I have addressed your issues that we have agreed need addressing.
